cmd/kubeadm/app/preflight/checks_test.go

```diff
@@ -185,124 +186,228 @@ func (pfct preflightCheckTest) Check() (warning, errorList []error) {
+			check: FileAvailableCheck{
+				Path: "/does/not/exist",
 			},
-			expected:                true,
-			isSecondaryControlPlane: true,
-			downloadCerts:           true,
+			expectedError: false,
 		},
 		{
-			name: "Test Cert file exists if specified",
-			cfg: &kubeadmapi.InitConfiguration{
-				ClusterConfiguration: kubeadmapi.ClusterConfiguration{
-					Etcd: kubeadmapi.Etcd{External: &kubeadmapi.ExternalEtcd{CertFile: "/foo"}},
-				},
+			name: "The file exists",
+			check: FileAvailableCheck{
+				Path: f.Name(),
 			},
-			expected: false,
+			expectedError: true,
 		},
+	}
+	for _, rt := range tests {
+		_, output := rt.check.Check()
+		if (output != nil) != rt.expectedError {
+			t.Errorf(
+				"Failed FileAvailableCheck:%v\n\texpectedError: %t\n\t  actual: %t",
+				rt.name,
+				rt.expectedError,
+				(output != nil),
+			)
+		}
+	}
+}
+
+func TestFileContentCheck(t *testing.T) {
+	f, err := os.CreateTemp("", "file-content-check")
+	if err != nil {
+		t.Fatalf("Failed to create file: %v", err)
+	}
+	defer os.Remove(f.Name())
+	var tests = []struct {
+		name          string
+		check         FileContentCheck
+		expectedError bool
+	}{
 		{
-			name: "Test Key file exists if specified",
-			cfg: &kubeadmapi.InitConfiguration{
-				ClusterConfiguration: kubeadmapi.ClusterConfiguration{
-					Etcd: kubeadmapi.Etcd{External: &kubeadmapi.ExternalEtcd{CertFile: "/foo"}},
-				},
+			name: "File exists and has matching content",
+			check: FileContentCheck{
+				Path:    f.Name(),
+				Content: []byte("Test FileContentCheck"),
 			},
-			expected: false,
+			expectedError: true,
 		},
 		{
-			name: "Test APIEndpoint exists if exists",
-			cfg: &kubeadmapi.InitConfiguration{
-				LocalAPIEndpoint: kubeadmapi.APIEndpoint{AdvertiseAddress: "2001:1234::1:15"},
+			name: "File exists, content is nil",
+			check: FileContentCheck{
+				Path:    f.Name(),
+				Content: nil,
 			},
-			expected: false,
+			expectedError: true,
 		},
+		{
+			name: "File exists but has unexpected content",
+			check: FileContentCheck{
+				Path:    f.Name(),
+				Content: []byte("foo"),
+			},
+			expectedError: false,
+		},
+		{
+			name: "File does not exist, content is not nil",
+			check: FileContentCheck{
+				Path:    "/does/not/exist",
+				Content: []byte("foo"),
+			},
+			expectedError: false,
+		},
+		{
+			name: "File dose not exist, content is nil",
+			check: FileContentCheck{
+				Path:    "/does/not/exist",
+				Content: nil,
+			},
+			expectedError: false,
+		},
+	}
+	if _, err = f.WriteString("Test FileContentCheck"); err != nil {
+		t.Fatalf("Failed to write to file: %v", err)
 	}
 	for _, rt := range tests {
-		// TODO: Make RunInitNodeChecks accept a ClusterConfiguration object instead of InitConfiguration
-		actual := RunInitNodeChecks(exec.New(), rt.cfg, sets.NewString(), rt.isSecondaryControlPlane, rt.downloadCerts)
-		if (actual == nil) != rt.expected {
+		_, output := rt.check.Check()
+		if (len(output) == 0) != rt.expectedError {
```

Comment:
i'm sorry but i think i made a mistake with earlier feedback.

the check here should be:
`if (len(output) > 0) != rt.expectedError {`

because `output > 0` means there are errors and `expectedError` means we are expecting errors.

(please remember to please keep the commits squashed to one)

Reply:
ok, did it.